sdk/src/cameras/3d-smart-camera/camera_3d_smart.cpp

```diff
@@ -0,0 +1,599 @@
+        if (status != Status::OK) {
+            LOG(WARNING) << "Failed to set frame type";
+            return status;
+        }
+        m_details.frameType = *frameDetailsIt;
+        //Turn on the streaming of data.
+        uint16_t afeRegsAddress[2] = {0x4001, 0x7c22};
+        uint16_t afeRegsValue[2] = {0x0007, 0x0004};
+        m_depthSensor->writeAfeRegisters(afeRegsAddress, afeRegsValue, 2);
+    }
+
+    if (!m_devStarted) {
+        status = m_depthSensor->start();
+        if (status != Status::OK) {
+            return status;
+        }
+        m_devStarted = true;
+    }
+
+    return status;
+}
+
+aditof::Status Camera3D_Smart::getAvailableFrameTypes(
+    std::vector<std::string> &availableFrameTypes) const {
+    using namespace aditof;
+    Status status = Status::OK;
+
+    std::vector<FrameDetails> frameDetailsList;
+    status = m_depthSensor->getAvailableFrameTypes(frameDetailsList);
+    if (status != Status::OK) {
+        LOG(WARNING) << "Failed to get available frame types";
+        return status;
+    }
+
+    for (const auto &item : frameDetailsList) {
+        availableFrameTypes.emplace_back(item.type);
+    }
+
+    return status;
+}
+
+aditof::Status
+Camera3D_Smart::requestFrame(aditof::Frame *frame,
+                             aditof::FrameUpdateCallback /*cb*/) {
+    using namespace aditof;
+    Status status = Status::OK;
+
+    if (frame == nullptr) {
+        LOG(ERROR) << "Received pointer frame is null";
+        return aditof::Status::INVALID_ARGUMENT;
+    }
+
+    FrameDetails frameDetails;
+    frame->getDetails(frameDetails);
+
+    if (m_details.frameType != frameDetails) {
+        frame->setDetails(m_details.frameType);
+    }
+
+    uint16_t *frameDataLocation;
+    frame->getData(FrameDataType::FULL_DATA, &frameDataLocation);
+
+    status = m_depthSensor->getFrame(frameDataLocation);
+    if (status != Status::OK) {
+        LOG(WARNING) << "Failed to get frame from device";
+        return status;
+    }
+
+    if (m_details.mode != skCustomMode &&
+        (m_details.frameType.type == "depth_ir" ||
+         m_details.frameType.type == "depth_only")) {
+        if (m_depthCorrection) {
+            m_calibration.calibrateDepth(frameDataLocation,
+                                         m_details.frameType.width *
+                                             m_details.frameType.height);
+        }
+        if (m_cameraGeometryCorrection) {
+            m_calibration.calibrateCameraGeometry(
+                frameDataLocation,
+                m_details.frameType.width * m_details.frameType.height);
+        }
+    }
+
+    return Status::OK;
+}
+
+aditof::Status
+Camera3D_Smart::getDetails(aditof::CameraDetails &details) const {
+    using namespace aditof;
+    Status status = Status::OK;
+
+    details = m_details;
+
+    return status;
+}
+
+std::shared_ptr<aditof::DepthSensorInterface> Camera3D_Smart::getSensor() {
+    return m_depthSensor;
+}
```

Comment:
Can we add both sensors here? or we will keep it only for the depthSensor?

Reply:
Yes we should return both. I will have to change the API and make changes for all the other cameras.

Reply:
fixed